homeassistant/components/starlink/sensor.py

```diff
@@ -0,0 +1,125 @@
+    """A sensor exposing Dishy's latency in milliseconds."""
+
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    _attr_native_unit_of_measurement = TIME_MILLISECONDS
+    _attr_icon = "mdi:speedometer"
+    _attr_unique_id = "ping"
+    _attr_name = "Ping"
+
+    @property
+    def native_value(self) -> float:
+        """Calculate the round-trip ping from coordinator data."""
+        raw_latency = self.coordinator.data.pop_ping_latency_ms
+        return round(raw_latency)
+
+
+class AzimuthSensor(StarlinkSensorEntity):
+    """A sensor exposing Dishy's Azimuth in degrees."""
+
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    _attr_native_unit_of_measurement = DEGREE
+    _attr_icon = "mdi:compass"
+    _attr_unique_id = "azimuth"
+    _attr_name = "Azimuth"
+
+    @property
+    def native_value(self) -> float:
+        """Get the azimuth from coordinator data."""
+        azimuth = self.coordinator.data.direction_azimuth
+        return azimuth
+
+
+class ElevationSensor(StarlinkSensorEntity):
+    """A sensor exposing Dishy's Elevation in degrees."""
+
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    _attr_native_unit_of_measurement = DEGREE
+    _attr_unique_id = "elevation"
+    _attr_name = "Elevation"
+
+    @property
+    def native_value(self) -> float:
+        """Get the elevation from coordinator data."""
+        elevation = self.coordinator.data.direction_elevation
+        return elevation
+
+
+class UplinkThroughputSensor(StarlinkSensorEntity):
+    """A sensor exposing Dishy's uplink throughput in bits per second."""
+
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    _attr_native_unit_of_measurement = DATA_RATE_MEGABITS_PER_SECOND
+    _attr_icon = "mdi:upload"
+    _attr_unique_id = "uplink_throughput"
+    _attr_name = "Uplink Throughput"
+
+    @property
+    def native_value(self) -> float:
+        """Calculate the uplink throughput based on coordinator data."""
+        raw_uplink_throughput = self.coordinator.data.uplink_throughput_bps
+        return round(raw_uplink_throughput / 1000000, 2)
+
+
+class DownlinkThroughputSensor(StarlinkSensorEntity):
+    """A sensor exposing Dishy's downlink throughput in bits per second."""
+
+    _attr_state_class = SensorStateClass.MEASUREMENT
+    _attr_native_unit_of_measurement = DATA_RATE_MEGABITS_PER_SECOND
+    _attr_icon = "mdi:download"
+    _attr_unique_id = "downlink_throughput"
+    _attr_name = "Downlink Throughput"
+
+    @property
+    def native_value(self) -> float:
+        """Calculate the downlink throughput based on coordinator data."""
+        raw_downlink_throughput = self.coordinator.data.downlink_throughput_bps
+        return round(raw_downlink_throughput / 1000000, 2)
+
+
+class LastBootTime(StarlinkSensorEntity):
+    """A sensor exposing Dishy's uptime."""
+
+    _attr_device_class = SensorDeviceClass.TIMESTAMP
+    _attr_unique_id = "last_boot_time"
+    _attr_name = "Last Boot Time"
```

Comment:
Since you use new naming scheme, you should use sentence case:

> Entity names should start with a capital letter, the rest of the words are lower case (unless it's a proper noun or a capitalized abbreviation of course)